lib/zlog_5424.c

```diff
@@ -605,12 +605,13 @@ static void gmtime_assafe(time_t ts, struct tm *res)
 	if (ts >= 306) /* Jan 1 of next year */
 		res->tm_year++;
 
-	static unsigned int months[13] = {
+	static time_t months[13] = {
 		0, 31, 61, 92, 122, 153, 184, 214, 245, 275, 306, 337, 365,
 	};
+	const size_t month_max = array_size(months) - 1;
 
-	for (size_t i = 0; i < array_size(months); i++) {
-		if ((unsigned int)ts < months[i + 1]) {
+	for (size_t i = 0; i < month_max; i++) {
+		if (ts < months[i + 1]) {
 			res->tm_mon = ((i + 2) % 12);
```

Comment:
I know you didn't change this line - but I'm not following it: why is `tm_mon`  being shifted in this way (why isn't January ... zero?)

Reply:
@mjstapp above these lines there is this comment:

```c
	/* note we're starting in march like the romans did... */
	if (ts >= 306) /* Jan 1 of next year */
		res->tm_year++;
```

So I think 0 is meant to be march and that would explain the sum + modulus.